core/commands/mv.py

```diff
@@ -0,0 +1,35 @@
+import functools
+import os
+import sublime
+from sublime_plugin import WindowCommand
+
+from ..git_command import GitCommand
+
+
+class GsMvCurrentFileCommand(WindowCommand, GitCommand):
+
+    """
+    Prompt the user for a new name for the current file.
+    """
+
+    def run(self):
+        if self.file_path:
+            parent, base_name = os.path.split(self.file_path)
+            on_done = functools.partial(self.on_done,
+                self.file_path, parent, base_name)
+            v = self.window.show_input_panel("New Name:", base_name,
+                on_done, None, None)
+            name, ext = os.path.splitext(base_name)
+            v.sel().clear()
+            v.sel().add(sublime.Region(0, len(name)))
+
+    def on_done(self, file_path, parent, base_name, new_name):
+        if new_name == base_name:
+            return
+        new_path = os.path.join(parent, new_name)
+
+        self.git("mv", file_path, new_path)
+        v = self.window.find_open_file(file_path)
+        if v:
+            v.retarget(new_path)
```

Comment:
Just use 
```python
v.run_command("select_all")
```

Reply:
I'm not sure I understand.  Is this referring to the selection code above?  If so, that's not the same.  It specifically is selecting the name without the extension.

Reply:
I think `retarget` is perfect for this usage.